R/calculate-effect-sizes.R

```diff
@@ -410,6 +411,35 @@ batch_calc_ES <- function(dat,
   
   ES_long_names <- names(ES_ests_long)
   
+  # weights
+  conditions <- dat %>% dplyr::select({{condition}}) %>% unique() %>% dplyr::pull()
+  
+  if (is.null(baseline_phase)) baseline_phase <- conditions[1]
+  
+  if (is.null(intervention_phase)) {
+    intervention_phase <- setdiff(conditions, baseline_phase)[1]
+    if (length(conditions) > 2) {
+      msg <- paste0("The 'condition' variable has more than two unique values. Treating '", intervention_phase, "' as the intervention phase.")
+      warning(msg)
+    }
+  }
+  
+  dat <- 
+    dat %>% 
+    dplyr::filter(!!rlang::sym(condition) %in% c(baseline_phase, intervention_phase)) %>% 
+    dplyr::mutate(!!rlang::sym(condition) := ifelse(!!rlang::sym(condition) == baseline_phase, "A", "B"))
+  
+  n_weights <- 

```

Comment:
Would it work to put this inside a conditional so that n_weights only gets calculated if needed based on the `weights` argument?